app/components/gleam-logo.ts

```diff
@@ -0,0 +1,134 @@
+  };
+}
+
+export default class GleamLogoComponent extends Component<GleamLogoSignature> {
+  animationInterval: number | null = null;
+  container: HTMLElement | null = null;
+  resizeObserver: ResizeObserver | null = null;
+  @tracked riveInstance: Rive | null = null;
+  @tracked containerSize: number = 0;
+
+  get containerStyle(): string {
+    const size = this.args.size || this.containerSize;
+
+    return `height: ${size}px; width: ${size}px; max-width: 100%; display: block;`;
+  }
+
+  @action
+  handleDidInsert(element: HTMLDivElement) {
+    this.container = element;
+
+    // Set up ResizeObserver if no size is provided
+    if (!this.args.size) {
+      // Get initial size from parent
+      const parentElement = element.parentElement;
+
+      if (parentElement) {
+        // Force a layout calculation to get accurate dimensions
+        const parentRect = parentElement.getBoundingClientRect();
+        this.containerSize = Math.min(parentRect.width, parentRect.height);
+
+        // Set up ResizeObserver to update the container size
+        this.resizeObserver = new ResizeObserver((entries) => {
+          for (const entry of entries) {
+            const newSize = Math.min(entry.contentRect.width, entry.contentRect.height);
+
+            if (newSize > 0) {
+              this.containerSize = newSize;
+            }
+          }
+        });
+
+        // Only observe the immediate parent element
+        this.resizeObserver.observe(parentElement);
+      }
+    }
+
+    try {
+      const canvas = document.createElement('canvas');
+
+      // Calculate base size based on device pixel ratio and container size
+      const pixelRatio = window.devicePixelRatio || 1;
+      const containerSize = this.args.size || this.containerSize;
+      // Use 2x the container size for high quality, but cap at 4x for performance
+      const qualityMultiplier = Math.min(4, Math.max(2, pixelRatio));
+      const baseSize = Math.round(containerSize * qualityMultiplier);
+
+      canvas.width = baseSize;
+      canvas.height = baseSize;
+
+      // Let the canvas scale naturally within its container
+      canvas.style.width = '100%';
+      canvas.style.height = '100%';
+      canvas.style.display = 'block';
+      canvas.style.maxWidth = '100%'; // Ensure it doesn't overflow on mobile
+
+      element.appendChild(canvas);
+
+      this.riveInstance = new Rive({
+        src: '/assets/animations/gleam_logo_animation.riv',
+        canvas: canvas,
+        stateMachines: 'Default',
+        autoplay: true,
+        automaticallyHandleEvents: true,
+        layout: new Layout({
+          fit: Fit.Contain,
+        }),
+        onLoad: async () => {
+          if (this.riveInstance) {
+            // Initial resize
+            this.riveInstance.resizeDrawingSurfaceToCanvas();
+          }
+        },
+      });
+    } catch (error: unknown) {
+      console.error('Error setting up Rive:', error);
+      Sentry.captureException(error, {
```

Comment:
Just `captureException` without extra tags should be fine @Arpan-206 - sentry should automatically capture context like component etc. Example: https://codecrafters.sentry.io/issues/6269915267/?project=5922961&query=is%3Aunresolved%20issue.priority%3A%5Bhigh%2C%20medium%5D&referrer=issue-stream&stream_index=5